### PR TITLE
Datepicker component

### LINKED ISSUE #231: Datepicker component
According to RFP proposal [specs](https://xd.adobe.com/view/44a3298e-a814-42b2-643e-3aa39c8dec34-3292/screen/73b5dadd-d93d-4f8f-b7b9-74c4ae8dd567/Create-RFP-proposal-/) we need a datepicker component which supports days for the RFP deadline input.

### PR DESCRIPTION
**Pull request by amassarwi**
closes #231 

Inspired by: https://github.com/nickeljew/react-month-picker - in addition we will add days support

### CODE REVIEW

**Comment by amassarwi** [comment]
Todos:
 - [ ] Convert styles to css modules
 - [ ] Smaller pad width in months mode
 - [ ] Create component for pad item - clickable `<li></li>` tags

File: src/components/Datepicker/Datepicker.jsx [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+  const handleClickMonth = useCallback((e) => {
+    const refid = getDID(e).split(":");
+    const idx = parseInt(refid[0], 10);
+    const month = parseInt(refid[1], 10);
+    const year = labelYearsState[idx];
+    const values = valuesState;
+    values[idx] = { year, month };
+    setValuesState(values);
+    onChange(year, month, idx);
+  }, []);
```
Comment (fernandoabolafio) [inline src/components/Datepicker/Datepicker.jsx]: Missing deps here

File: src/components/Datepicker/Datepicker.jsx [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+      _setYear(idx, -1);
+    }
+  }, []);
+
+  const handleNextYearClick = useCallback((e) => {
+    const idx = parseInt(getDID(e), 10);
+    if (yearIndexesState[idx] < yearsState.length - 1) {
+      _setYear(idx, 1);
+    }
+  }, []);
```
Comment (fernandoabolafio) [inline src/components/Datepicker/Datepicker.jsx]: Missing deps here

**Comment by tiagoalvesdulce** [comment]
Do you think we can disable some months for a given year? This would come handy in cms. For example, we are now in March and it doesn't make sense allowing users to select months other than January and February.

**Comment by amassarwi** [comment]
@tiagoalvesdulce 
It's there, you can specify min & max and months will be disables accordingly:
See: https://github.com/decred/politeiagui/pull/1825 already has it working.

I'll added to month docz example

**Comment by tiagoalvesdulce** [comment]
Yeah, add it to the docs then. Good job!

**Comment by amassarwi** [comment]
@tiagoalvesdulce added test & updated docz example

**Pull request merged**